160647449 Users should be able to sign up endpoint

V-84 opened the pull request:
#### What does this PR do?
sign up users endpoint
#### Description of Task to be completed?
- created a sign up route
- created a  userSignUp controller
- optimized userValidation middleware
#### How should this be manually tested?
screenshots
#### Any background context you want to provide?
n/a
#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/n/projects/2196426/stories/160647449
#### Screenshots (if appropriate)

![screenshot from 2018-09-23 08-31-51](https://user-images.githubusercontent.com/36252436/45925559-aba60900-bf0f-11e8-9a42-142e80138da3.png)
![screenshot from 2018-09-23 08-32-02](https://user-images.githubusercontent.com/36252436/45925560-ac3e9f80-bf0f-11e8-8481-324d772446da.png)

#### Questions: n/a

V-84 committed:
feature: sign up endpoint
- created a sign up route
- created a  userSignUp controller
- optimized userValidation middleware

[delivers 160647449]

fob413 commented on server/controllers/users.js:
```diff
@@ -1 +1,78 @@
+    const { email, username, password, deliveryAddress, phoneNo, firstname, lastname, imageUrl } = req.body;
+    bcrypt.genSalt(10, (err, salt) => {
+      bcrypt.hash(password, salt, (err, hash) => {
+        db.one('INSERT INTO users(firstname, lastname,phoneNo,deliveryAddress,imageurl, usertype, username,password, email) values($1,$2,$3,$4,$5,$6,$7,$8,$9) RETURNING id,username,imageurl,firstname, userType', [firstname, lastname, phoneNo, deliveryAddress, imageUrl, userType, username, hash, email])
```
Does your `eslint` flag this as an error? If it does, please fix this and any other line that the linter flags

fob413 commented:
Please go through the recomended git workflow and update your PR name

V-84 committed:
removed comments
[delivers #160647449]

V-84 committed:
implement feedback
- shorten the length of the expression in users.js
- wrote the test for signup
- removed comments

V-84 committed:
updated travis.yml file

V-84 committed:
updated postgres password on travis
